compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -161,7 +161,9 @@ class SyntheticMethods(thisPhase: DenotTransformer) {
       val thatAsClazz = ctx.newSymbol(ctx.owner, nme.x_0, Synthetic, clazzType, coord = ctx.owner.pos) // x$0
       def wildcardAscription(tp: Type) = Typed(Underscore(tp), TypeTree(tp))
       val pattern = Bind(thatAsClazz, wildcardAscription(AnnotatedType(clazzType, Annotation(defn.UncheckedAnnot)))) // x$0 @ (_: C @unchecked)
-      val comparisons = accessors map { accessor =>
+      //compare primitive fields first, slow equality checks of non-primitive fields can be skipped when primitives differ
+      val accessorsParts = accessors.partition(_.info.typeSymbol.isPrimitiveValueClass)
```

Comment:
You should be able to use `_.classSymbol.isPrimitiveValueClass` instead of `_.info...`
But how about something like:
```scala
accessors.sortBy(acc => if (acc.classSymbol.isPrimitiveValueClass) 0 else 1)
```
Sorting is stable so this won't reorder two fields which are both primitive or both non-primitive.

Reply:
Do you mean `_.info.classSymbol.isPrimitiveValueClass`? `accessor` is `dotty.tools.dotc.core.Symbols.Symbol`, which doesn't seem to have `classSymbol`

Reply:
ah indeed, then you can keep doing info.typeSymbol yeah.